index.ts

```diff
@@ -12,6 +12,18 @@ type Attributes = JSX.IntrinsicElements['div'];
 type DocumentFragmentConstructor = typeof DocumentFragment;
 type ElementFunction = () => HTMLElement | SVGElement;
 
+declare global {
+	namespace JSX {
+		interface Element extends HTMLElement {}
```

Comment:
This also needs `SVGElement` and maybe also `DocumentFragment` as used here:

https://github.com/sindresorhus/refined-github/blob/62b69a571f9e43c07b7e18911226e4f6467b9829/source/globals.d.ts#L47-L48

Reply:
i get an error if i add `SVGElement` and/or `DocumentFragment` because of differences in the `addEventListener` declaration, the reason why there are no issues in `refined-github/source/globals.d.ts` is because for some reason you don't get the errors in `d.ts` files, but moving the declaration to the `global.d.ts` file results in more errors, so i would say we should leave it as it is and deal with potential error later.

Reply:
Then what happens if you run `document.body.append(<svg/>)`?

`dom-chef` is supposed to be compatible with HTML, SVG and `<>`, so the solution we have to choose is one that supports both, even if it then forces you to write `(<div/> as unknown as HTMLDivElement).addEventListener`

Reply:
this works `document.body.appendChild(<svg></svg>);`
```ts
  const svg = <svg></svg>
  const div = <div></div>
```
the `svg` and `div` variables have the same type which is `HTMLElement + ReactElement?`, maybe we should change the type based on what tag it is but i don't know how to do that, its probably possible but it would requirie reading the soure code, and the current solutiuion is simple and works.

Reply:
this `(<></>)` doesn't work and won't work since its a typescript problem and not a `dom-chef` problem.

Reply:
`<>` works in TSX: https://github.com/sindresorhus/refined-github/blob/13ed4a8ef04b3b9490aa485a27ad077a10fdd32f/source/options.tsx#L29

We [mention](https://github.com/vadimdemedes/dom-chef#alternative-usage) that in the readme (however that's not super clear)

I think it needs to extend the 3 types or else this won't compile:

```tsx
function a(): SVGElement {
	return <svg viewBox="0 0 1 1">
}
function b(): DocumentFragment {
	return <>b</>
}
function c(): HTMLElement {
	return <div />
}
```

This does make the types looser:


```tsx
function a(): DocumentFragment { /* any of the 3 types works, looser*/
	return <svg viewBox="0 0 1 1">
}
function b(): HTMLElement {
	return <>b</>
}
function c(): SVGElement {
	return <div />
}
```


But I think it's better than the alternative:


```tsx
function a(): HTMLElement { /* only HTMLElement works, wrong */
	return <svg viewBox="0 0 1 1">
}
function b(): HTMLElement {
	return <>b</>
}
function c(): HTMLElement {
	return <div />
}
```

Reply:
the latest commit extends `JSX.Element` further, the example should compile, i got a bunch of error in my clean test project but they are all from `dom-chef` (probably because of yarn link) so i think it should work now.